Assets/Scripts/View/ToolBar/Timer.cs

```diff
@@ -22,42 +25,49 @@ public IEnumerator StartTimer(int time)
         {
             CurrentTime = time;
             GameState state = controller.game.State.CurrentState;
+            var last_tick = PhotonNetwork.Time;
             while (CurrentTime > 0 && state == controller.game.State.CurrentState)
             {
                 yield return null;
-                CurrentTime -= Time.deltaTime;
+                var now = PhotonNetwork.Time;
+                CurrentTime -= (float)(now - last_tick);
+                last_tick = now;
             }
             CurrentTime = 0f;
             if (state == controller.game.State.CurrentState)
             {
+                PhotonNetwork.RaiseEvent(12, null, controller.RaiseEventOptions, controller.SendOptions);
                 switcher.GoCommon();
-                foreach (Player player in controller.Players)
-                    player.GetReady();
+                controller.MinePlayer.GetReady();
             }
         }
     }
 
     private void Update()
     {
-        if (!controller.IsReadyToGoNext && !controller.AllPlayersReady && CurrentTime == 0)
+        if (PhotonNetwork.IsMasterClient)
         {
-            int AimTime = 0;
-            switch (controller.game.State.CurrentState)
+            if (!controller.IsReadyToGoNext && !controller.AllPlayersReady && CurrentTime == 0)
             {
-                case GameState.FixCosts:
-                    AimTime = 60 + 60;
-                    break;
-                case GameState.MatRequest:
-                    AimTime = 60 + 60;
-                    break;
-                case GameState.ProdRequest:
-                    AimTime = 60 + 60;
-                    break;
-                case GameState.Production:
-                    AimTime = 60 + 60;
-                    break;
+                int AimTime = 0;
+                switch (controller.game.State.CurrentState)
+                {
+                    case GameState.BuildUpgrade:
+                        AimTime = 60 + 60;
```

Comment:
Why it's `60 + 60`?

Reply:
To see how many minutes (random, but how I understand this time fits perfect).

Reply:
I see. I would use `AimTimeInMinutes = 2 * 60`, or, even better, it's better to create a constant. 

Generally, it's better to avoid any magic numbers! https://en.wikipedia.org/wiki/Magic_number_(programming)